src/codecs/decoding/format/syslog.rs

```diff
@@ -20,6 +22,30 @@ impl SyslogDeserializerConfig {
     pub const fn build(&self) -> SyslogDeserializer {
         SyslogDeserializer
     }
+
+    /// The schema produced by the deserializer.
+    pub fn schema_definition(&self) -> schema::Definition {
+        schema::Definition::empty()
+            // The `message` field is always defined. If parsing fails, the entire body becomes the
+            // message.
+            .required_field(log_schema().message_key(), Kind::bytes(), Some("message"))
```

Comment:
Note to self: this is no longer true, since #11244 got merged as I worked on getting this PR up. I'll revisit this.